### PR TITLE
Interfaces can now have GUI changes

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Adds a custom class to have GUI changes on the inspector from interfaces.
Transforms the SubscriptionController into an interface.
Removes the Editor.asmdef
CL: [Fix] Fixed the door button linking missing from the inspector.

### CODE REVIEW

**Comment by Aranclanos** [comment]
Draft because of some renaming and warning spam, plus I believe theres a third interface that needs to be added into this

File: UnityProject/Assets/Scripts/Systems/Electricity/Inheritance/SubscriptionController.cs [hunk 1]
```diff
@@ -2,19 +2,18 @@
...
 using Mirror;
 
-
 namespace Systems.ObjectConnection
 {
 	/// <summary>
 	/// Used for editor scripts.
 	/// </summary>
-	public abstract class SubscriptionController : NetworkBehaviour
+	public interface SubscriptionController
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Systems/Electricity/Inheritance/SubscriptionController.cs]:
```suggestion
	public interface ISubscriptionController
```

File: UnityProject/Assets/Scripts/Core/Editor/Inspector/InterfaceGUI.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+using Systems.ObjectConnection;
+using Mirror;
+using UnityEditor;
+using UnityEngine;
+using System.Collections.Generic;
+
+namespace CustomInspectors
+{
+	[ExecuteInEditMode]
+	public class InterfaceGUI : NetworkBehaviour
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Editor/Inspector/InterfaceGUI.cs]: name is way too generic for its very specific use which is rendering the interface for multitool masters

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Core/Editor/Inspector/InterfaceGUI.cs]: and door buttons! and a third one later

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Editor/Inspector/InterfaceGUI.cs]: door buttons are as well using the multitool subscription system, aren't they?

**Comment by Aranclanos** [comment]
no longer a draft

**Comment by NoooneyDude** [comment]
Tests are sad

**Comment by ThatDan123** [comment]
Why did you need to remove the editor Asmdef, i think not having it will cause builds to fail as it keeps scripts with editor only namespaces outside the build.

File: UnityProject/Assets/Scripts/Core/Editor/Inspector/MasterDeviceInspector.cs [hunk 3]
```diff
@@ -27,13 +19,14 @@ public class MasterDeviceInspector : Editor
...
+		public override void OnEnableInEditor(object target)
 		{
-			DeviceLinker.InitDeviceLists(((IMultitoolMasterable) target).ConType, forceRefresh: true);
+			var asd = (IMultitoolMasterable)target;
+			DeviceLinker.InitDeviceLists(asd.ConType, true);
 		}
 
 		[DrawGizmo(GizmoType.Selected | GizmoType.Active)]
-		private static void DrawGizmoConnection(IMultitoolMasterable device, GizmoType type)
+		public void DrawGizmoConnectionCALL(IMultitoolMasterable device, GizmoType type)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Editor/Inspector/MasterDeviceInspector.cs]: This needs to be static for `[DrawGizmo]`.

https://docs.unity3d.com/ScriptReference/DrawGizmo.html

File: UnityProject/Assets/Scripts/Core/Editor/Inspector/InterfaceGUI.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		private void OnDisable()
+		{
+			foreach (var interfaceEntry in interfaceGUI.runningInterfaces)
+			{
+				interfaceEntry.OnDisableInEditor(target);
+			}
+		}
+
+		[DrawGizmo(GizmoType.Selected | GizmoType.Active)]
+		private void DrawGizmoConnection(IMultitoolMasterable device, GizmoType type)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Editor/Inspector/InterfaceGUI.cs]: Needs to be static here too.

**Review by corp-0 — DISMISSED** [review]

**Review by corp-0 — DISMISSED** [review]

**Comment by Bod9001** [comment]
just need to wait tests

**Review by corp-0 — APPROVED** [review]

**Pull request merged**